samples/sync-logins/src/main/java/org/mozilla/samples/sync/logins/MainActivity.kt

```diff
@@ -0,0 +1,171 @@
+        //const val CONFIG_URL = "https://latest.dev.lcip.org"
+        const val CONFIG_URL = "https://accounts.firefox.com"
+        const val FXA_STATE_PREFS_KEY = "fxaAppState"
+        const val FXA_STATE_KEY = "fxaState"
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+
+        listView = findViewById<ListView>(R.id.logins_list_view)
+        adapter = ArrayAdapter(this, android.R.layout.simple_list_item_1)
+        listView.adapter = adapter
+        activityContext = this;
+
+        getSharedPreferences(FXA_STATE_PREFS_KEY, Context.MODE_PRIVATE).getString(FXA_STATE_KEY, "").let {
+            FirefoxAccount.fromJSONString(it).then({
+                this.account = it
+                account?.getProfile()
+            }, {
+                Config.custom(CONFIG_URL).whenComplete { value: Config ->
+                    account = FirefoxAccount(value, CLIENT_ID, REDIRECT_URL)
+                }
+                FxaResult()
+            }).whenComplete {
+                val txtView: TextView = findViewById(R.id.txtView)
+                runOnUiThread {
+                    txtView.text = getString(R.string.signed_in, "${it.displayName
+                            ?: ""} ${it.email}")
+                }
+            }
+        }
+
+        findViewById<View>(R.id.buttonWebView).setOnClickListener {
+            account?.beginOAuthFlow(scopes, wantsKeys)?.whenComplete { openWebView(it) }
+        }
+
+        findViewById<View>(R.id.buttonLogout).setOnClickListener {
+            getSharedPreferences(FXA_STATE_PREFS_KEY, Context.MODE_PRIVATE).edit().putString(FXA_STATE_KEY, "").apply()
+            val txtView: TextView = findViewById(R.id.txtView)
+            txtView.text = getString(R.string.logged_out)
+        }
+
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        account?.close()
+    }
+
+    override fun onNewIntent(intent: Intent) {
+        super.onNewIntent(intent)
+        val action = intent.action
+        val data = intent.dataString
+
+        if (Intent.ACTION_VIEW == action && data != null) {
+            val url = Uri.parse(data)
+            val code = url.getQueryParameter("code")
+            val state = url.getQueryParameter("state")
+            displayAndPersistProfile(code, state)
+        }
+    }
+
+    private fun openWebView(url: String) {
+        supportFragmentManager?.beginTransaction()?.apply {
+            replace(R.id.container, LoginFragment.create(url, REDIRECT_URL))
+            addToBackStack(null)
+            commit()
+        }
+    }
+
+    override fun onLoginComplete(code: String, state: String, fragment: LoginFragment) {
+        displayAndPersistProfile(code, state)
+        supportFragmentManager?.popBackStack()
+    }
+
+    fun dumpError(tag: String, e: Exception) {
+        val sw = StringWriter();
```

Comment:
Let's clean this up use the a-c Logger: https://github.com/mozilla-mobile/android-components/blob/master/components/support/base/src/main/java/mozilla/components/support/base/log/logger/Logger.kt